rocks/RocksIterator.java

```diff
@@ -81,18 +83,26 @@ public final boolean hasNext() {
             case EMPTY:
                 return fetchAndCheck();
             case INIT:
-                initalise();
-                return fetchAndCheck();
+                return initialise();
             default: // This should never be reached
                 return false;
         }
     }
 
-    @Override
-    public final T next() {
-        if (!hasNext()) throw new NoSuchElementException();
-        state = State.EMPTY;
-        return next;
+    private synchronized boolean fetchAndCheck() {
```

Comment:
Do we need to reorder this method in this class? It's harder to review the correctness of this PR now as more things change than necessary.